api/src/server.rs

```diff
@@ -399,7 +407,17 @@ where
             // rely on the channel depth for backpressure. the goal is to keep the queue close to empty
             // without processing one at a time. when we stop parsing the carfile in the store
             // i.e. validate before sending here and this is just an insert, we may want to process more at once.
+            let mut shutdown = false;
             loop {
+                let should_exit = shutdown || event_rx.is_closed();
+                // make sure the events queue doesn't get too deep when we're under heavy load
```

Comment:
nitpick, it confused me to have this code come in the middle of the shutdown logic.  Can we put line 412 and the block starting on line 417 next to each other?  Ideally as the first thing in the loop?

Reply:
reworked the logic a bit to stop using channel openness at all or share anything between shutdown and normal processing. I added draining the channel before shutting down. lmk what you think.